lib/utils/getContract/index.ts

```diff
@@ -0,0 +1,63 @@
+import { ethers, providers, Signer } from "ethers";
+
+import { addresses } from "../../constants";
+import IERC20 from "../../abi/IERC20.json";
+import WSKLIMA from "../../abi/wsKlima.json";
+import DistributorContractv4 from "../../abi/DistributorContractv4.json";
+import PunkTLD from "../../abi/PunkTLD.json";
+import PairContract from "../../abi/PairContract.json";
+import KlimaRetirementAggregator from "../../abi/KlimaRetirementAggregator.json";
+import ExercisePKlima from "../../abi/ExercisepKLIMA.json";
+import KlimaStakingHelper from "../../abi/KlimaStakingHelper.json";
+import KlimaStakingv2 from "../../abi/KlimaStakingv2.json";
+import KlimaRetirementStorage from "../../abi/KlimaRetirementStorage.json";
+
+type Token = keyof typeof addresses["mainnet"];
+type ContractMap = {
+  [K in Token]: typeof IERC20["abi"] | typeof KlimaStakingHelper["abi"];
+};
+const contractMap = {
```

Comment:
i think this is a good enough solution

if we do decide to make a single-source of truth config, that would be ideal, but that's also a big refactor because it would lead to us refactoring the `addresses` object entirely in `@klimadao/lib/constants/index.tsx`

we use `addresses.mainnet` in soo many places. So for the sake of avoiding the refactor dominos lets revisit later?
![image](https://user-images.githubusercontent.com/88635679/174926466-52e3ec5c-8be9-4345-aabc-f26df26d0690.png)

Reply:
Yes, let´s do this in a follow-up
This PR is already quite delicate.